### PR TITLE
mam/v2: sponge - decision is to stay with trit_t

### PR DESCRIPTION
**Pull request by tsvisabo**
- Revert to trits as it was decided we keep that type and for performance it is best that sponge transform 
  function will operate on "plain" trits

### CODE REVIEW

File: mam/v2/curl.c [hunk 1]
```diff
@@ -23,28 +23,17 @@ static trit_t curl_table[] = {1, 0, -1, 2, 1, -1, 0, 2, -1, 1, 0};
...
+  trit_t t, prev_t = trits_get1(state);
   size_t i, j = 0;
-
-  flex_trit_t *stack_p = flex_trits_stack;
-  size_t stack_pos = 0;
-  size_t state_pos = 0;
   for (; rounds--;) {
-    flex_trits_slice(flex_trits_state, FLEX_SPONGE_WIDTH, flex_trits_stack,
-                     FLEX_SPONGE_WIDTH, state_pos, FLEX_SPONGE_WIDTH);
+    trits_copy(x = state, stack);
```
Comment (thibault-martinez) [inline mam/v2/curl.c]: Can you put this assignment outside ?

File: mam/v2/curl.c [hunk 2]
```diff
@@ -23,28 +23,17 @@ static trit_t curl_table[] = {1, 0, -1, 2, 1, -1, 0, 2, -1, 1, 0};
...
-  flex_trit_t flex_trits_state[FLEX_SPONGE_WIDTH];
-  flex_trits_from_trits(flex_trits_state, FLEX_SPONGE_WIDTH, s,
-                        FLEX_SPONGE_WIDTH, FLEX_SPONGE_WIDTH);
-  flex_trit_t flex_trits_stack[FLEX_SPONGE_WIDTH];
-  flex_trits_from_trits(flex_trits_stack, FLEX_SPONGE_WIDTH, (trit_t *)buf,
-                        FLEX_SPONGE_WIDTH, FLEX_SPONGE_WIDTH);
-  trit_t t, prev_t = flex_trits_at(flex_trits_state, FLEX_SPONGE_WIDTH, 0);
+  trits_t x, state = trits_from_rep(MAM2_SPONGE_WIDTH, s);
+  trits_t stack = trits_from_rep(MAM2_SPONGE_WIDTH, (trit_t *)buf);
+  trit_t t, prev_t = trits_get1(state);
```
Comment (thibault-martinez) [inline mam/v2/curl.c]: Could you rename `prev_t` to simply `prev` to avoid type confusion ?

File: mam/v2/prng.c [hunk 3]
```diff
@@ -36,10 +36,23 @@ MAM2_SAPI void prng_init(iprng *p, isponge *s, trits_t K) {
   trits_copy(K, prng_key_trits(p));
 }
 
+static void flex_from_trits(trits_t t, flex_trit_t *flex_trits) {
```
Comment (thibault-martinez) [inline mam/v2/prng.c]: Put this one in common ?

File: mam/v2/tests/common.h [hunk 4]
```diff
@@ -16,6 +16,7 @@
 #ifndef __MAM_V2_TEST_STRUCTS_H__
 #define __MAM_V2_TEST_STRUCTS_H__
 
+#include "../trits.h"
```
Comment (thibault-martinez) [inline mam/v2/tests/common.h]: Avoid relative paths

File: mam/v2/trits.h [hunk 5]
```diff
@@ -19,10 +19,10 @@
 #ifndef __MAM_V2_TRITS_H__
 #define __MAM_V2_TRITS_H__
 
-#include "mam/v2/alloc.h"
-#include "mam/v2/defs.h"
+#include "alloc.h"
+#include "defs.h"
 
-#include "common/trinary/trit_tryte.h"
+#include "../../common/trinary/trit_tryte.h"
```
Comment (thibault-martinez) [inline mam/v2/trits.h]: Relative path

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

File: mam/v2/trits.h (hunk 5 above)
Comment (tsvisabo) [reply mam/v2/trits.h]: CLion...

File: mam/v2/prng.c (hunk 3 above)
Comment (tsvisabo) [reply mam/v2/prng.c]: for now it will be in trits.h

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**